sky/tools/create_ios_framework.py

```diff
@@ -80,20 +86,42 @@ def main():
   ])
   process_framework(args, fat_framework, fat_framework_binary)
 
-  # Create XCFramework from the arm-only fat framework and the simulator framework.
+  if args.simulator_arm64_out_dir is not None:
+    shutil.rmtree(fat_simulator_framework, True)
+    shutil.copytree(simulator_arm64_framework, fat_simulator_framework)
+
+    fat_simulator_framework_binary = os.path.join(fat_simulator_framework, 'Flutter')
+
+    # Create the arm64/x64 simulator fat framework.
+    subprocess.check_call([
+      'lipo',
+      simulator_x64_dylib,
+      simulator_arm64_dylib,
+      '-create',
+      '-output',
+      fat_simulator_framework_binary
+    ])
+    process_framework(args, fat_simulator_framework, fat_simulator_framework_binary)
+    simulator_framework = fat_simulator_framework
+  else:
+    simulator_framework = simulator_x64_framework
+
+  # Create XCFramework from the arm-only fat framework and the arm64/x64 simulator frameworks, or just the
+  # x64 simulator framework if only that one exists.
   xcframeworks = [simulator_framework, fat_framework]
   create_xcframework(location=args.dst, name='Flutter', frameworks=xcframeworks)
 
-  # Add the simulator into the fat framework.
+  # Add the x64 simulator into the fat framework
```

Comment:
The fat framework is [not published on its own outside the XCFramework anymore](https://flutter-review.googlesource.com/c/recipes/+/9560/3/recipes/engine.py), so if you want you can remove this section (everything after `create_xcframework`).


Or you could do it with your `--simulator-out-dir` cleanup later.

Reply:
Let's keep it for now, we can do cleanup later. I don't want to remove functionality just in case it breaks something downstream.